src/main/java/pl/gov/coi/pomocua/ads/myoffers/MyOffersResource.java

```diff
@@ -0,0 +1,33 @@
+package pl.gov.coi.pomocua.ads.myoffers;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import pl.gov.coi.pomocua.ads.BaseOffer;
+import pl.gov.coi.pomocua.ads.UserId;
+import pl.gov.coi.pomocua.ads.authentication.CurrentUser;
+
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/secure/")
+public class MyOffersResource {
+    private final MyOffersRepository repository;
+    private final CurrentUser currentUser;
+
+    @GetMapping("my-offers")
+    public Page<BaseOffer> list(Pageable pageRequest) {
+        UserId userId = currentUser.getCurrentUserId();
+        return repository.findAllByUserId(userId, pageRequest);
+    }
+
+    @GetMapping("my-offers/{id}")
+    public Optional<BaseOffer> get(@PathVariable Long id) {
+        return repository.findById(id);
```

Comment:
should also check if the offer with this id belongs to `currentUser`. Switch to `findOneByIdAndUserId`?

Reply:
and it also should be wrapped with `ResponseEntity.of(...)`, otherwise, when empty Optional is returned, it's HTTP 200 with body "null" (and we want it to be HTTP 404)